src/main/java/org/kohsuke/github/GHMyself.java

```diff
@@ -109,16 +121,31 @@
     }
 
     /**
-     * Lists up all the repositories this user owns (public and private) using the specified page size.
+     * List repositories that are accessible to the authenticated user (public and private) using the specified page size.
+     *
+     * This includes repositories owned by the authenticated user, repositories that belong to other users
+     * where the authenticated user is a collaborator, and other organizations' repositories that the authenticated
+     * user has access to through an organization membership.
      *
      * @param pageSize size for each page of items returned by GitHub. Maximum page size is 100.
      *
      * Unlike {@link #getRepositories()}, this does not wait until all the repositories are returned.
      */
     public PagedIterable<GHRepository> listRepositories(final int pageSize) {
+        return listRepositories(pageSize, RepositoryType.All);
+    }
+
+    /**
+     * List repositories of a certain type that are accessible by current authenticated user using the specified page size.
+     *
+     * @param pageSize size for each page of items returned by GitHub. Maximum page size is 100.
+     * @param repoType type of repository returned in the listing
+     */
+    public PagedIterable<GHRepository> listRepositories(final int pageSize, final RepositoryType repoType) {
         return new PagedIterable<GHRepository>() {
             public PagedIterator<GHRepository> iterator() {
-                return new PagedIterator<GHRepository>(root.retrieve().asIterator("/user/repos?per_page=" + pageSize, GHRepository[].class)) {
+                return new PagedIterator<GHRepository>(root.retrieve().asIterator("/user/repos?per_page=" + pageSize +
+                    "&type=" + repoType.toString().toLowerCase(), GHRepository[].class)) {
```

Comment:
also, can use `String.format("/user/repos?per_page=%s&type=%s", pageSize,  repoType)` if you override toString in enum